src/services/codefixes/fixMissingTypeAnnotationOnExports.ts

```diff
@@ -234,16 +238,61 @@ function withChanges<T>(
 
     function addFullAnnotation(span: TextSpan) {
         const nodeWithDiag = getTokenAtPosition(sourceFile, span.start);
-        const nodeWithNoType = findNearestParentWithTypeAnnotation(nodeWithDiag) ?? findExpandoFunction(nodeWithDiag);
+        const expandoFunction = findExpandoFunction(nodeWithDiag);
+        if (expandoFunction) {
+          if (isFunctionDeclaration(expandoFunction)) {
+            return createNamespaceForExpandoProperties(expandoFunction);
+          }
+          return fixupForIsolatedDeclarations(expandoFunction);
+        }
+        const nodeWithNoType = findNearestParentWithTypeAnnotation(nodeWithDiag)
         if (nodeWithNoType) {
             return fixupForIsolatedDeclarations(nodeWithNoType);
         }
         return undefined;
     }
 
+    function createNamespaceForExpandoProperties(expandoFunc: FunctionDeclaration): DiagnosticOrDiagnosticAndArguments|undefined {
+      if (expandoPropertiesAdded?.has(expandoFunc)) return undefined;
+      expandoPropertiesAdded?.add(expandoFunc);
+      const type = typeChecker.getTypeAtLocation(expandoFunc);
+      const elements = typeChecker.getPropertiesOfType(type);
+      if (!expandoFunc.name || elements.length === 0) return undefined;
+      const newProperties = []
+      for (const symbol of elements) {
+        if (!isIdentifierText(symbol.name, program.getCompilerOptions().target)) continue;
+        newProperties.push(factory.createVariableStatement(
+          /*modifiers*/ undefined,

```

Comment:
Shouldn't the variable declaration be exported? If the variable is not exported from the namespace it will not be the same as the one that gets added through the assignment 

```ts

export function foo() {}
foo.x = 1;
foo.y = 1;

export namespace foo {
    let x: string// not the same x as the one outside
}
```